src/vulkan-renderer/vk_tools/gpu_info.cpp

```diff
@@ -338,28 +338,28 @@ void print_physical_device_memory_properties(const VkPhysicalDevice graphics_car
 
         const auto &property_flag = gpu_memory_properties.memoryTypes[i].propertyFlags;
 
-        if (property_flag & VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT)) {
             spdlog::debug("VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT");
         }
-        if (property_flag & VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT)) {
             spdlog::debug("VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT");
         }
-        if (property_flag & VK_MEMORY_PROPERTY_HOST_COHERENT_BIT) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_PROPERTY_HOST_COHERENT_BIT)) {
             spdlog::debug("VK_MEMORY_PROPERTY_HOST_COHERENT_BIT");
         }
-        if (property_flag & VK_MEMORY_PROPERTY_HOST_CACHED_BIT) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_PROPERTY_HOST_CACHED_BIT)) {
             spdlog::debug("VK_MEMORY_PROPERTY_HOST_CACHED_BIT");
         }
-        if (property_flag & VK_MEMORY_PROPERTY_LAZILY_ALLOCATED_BIT) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_PROPERTY_LAZILY_ALLOCATED_BIT)) {
             spdlog::debug("VK_MEMORY_PROPERTY_LAZILY_ALLOCATED_BIT");
         }
-        if (property_flag & VK_MEMORY_PROPERTY_PROTECTED_BIT) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_PROPERTY_PROTECTED_BIT)) {
             spdlog::debug("VK_MEMORY_PROPERTY_PROTECTED_BIT");
         }
-        if (property_flag & VK_MEMORY_PROPERTY_DEVICE_COHERENT_BIT_AMD) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_PROPERTY_DEVICE_COHERENT_BIT_AMD)) {
             spdlog::debug("VK_MEMORY_PROPERTY_DEVICE_COHERENT_BIT_AMD");
         }
-        if (property_flag & VK_MEMORY_PROPERTY_DEVICE_UNCACHED_BIT_AMD) {
+        if (static_cast<bool>(property_flag & VK_MEMORY_PROPERTY_DEVICE_UNCACHED_BIT_AMD)) {
             spdlog::debug("VK_MEMORY_PROPERTY_DEVICE_UNCACHED_BIT_AMD");
         }
     }
```

Comment:
When discussing those lines with you, you mentioned that "most of them shouldn't actually be static_casts", but `!= 0`.